lib/rules/no-else-return.js

```diff
@@ -189,13 +198,14 @@ module.exports = {
             return checkForReturnOrIf(node);
         }
 
+
         /**
-         * Check the if statement
+         * Check the if statement, but don't catch else-if blocks.
          * @returns {void}
          * @param {Node} node The node for the if statement to check
          * @private
          */
-        function IfStatement(node) {
+        function LegacyIfStatement(node) {
```

Comment:
Can you give these functions more descriptive names? For example, `checkIfWithElse` and `checkIfWithoutElse` could work. I don't think using the word "legacy" makes sense here.